docs/roadmap.md

```diff
@@ -0,0 +1,185 @@
+# Roadmap
+
+## Aim of the project
+
+* Enable application developers to connect their workloads to backing services **with little or no code changes**.
+
+* Enable backing service authors to expose binding information by adding metadata to the the representative Kubernetes objects **with little or no code changes**.
+
+* Provide a managed binding experience.
+
+The Service Binding Operator is being designed to be adopted by application developers and backing service providers with a **low barrier for entry**.
+
+
+# Key feature areas
+
+### Collection of binding information
+* Annotation "Decorators" in backing service resources.
+* Descriptor "Decorators" in backing service resources.
+* Duck type in backing service resources.
+* Service Composition.
+* Detection of binding resources.
+
+### Project of binding informattion
+* Custom binding variable generation.
+* Projection of binding information in workloads.
+* Projection in podSpec and non-PodSpec based workloads.
+
+### Security
+* Avoiding escalation of privilege.
+* Narrow down the service account privileges.
+
+### Release & Packaging
+* Packaging of the Service Binding Controller/CRD.
+
+----
+
+
+# Collection of binding information
+
+## Annotation "Decorators" in backing service resources
+
+Supporting annotations as a way to decorate kubernetes resources enables the following:
+
+* Application developers can use any kubernetes resource a potential backing service
+* Backing service providers can annotate their CRDs to indicate what needs to be extracted for binding.
+
+The project aims to support an annotation format such that there's a way to indicate whether a specific unit of binding information is to be bound as a volume mount or an environment variable.
+
+## Improvements
+
+Validate advanced volume mounting support for projecting sensitive information as "files".
+
+
+## OLM Descriptor "Decorators" in backing service resources
+
+Use of OLM Descriptors as a way to indicate what is interesting for binding enables operator-backed services to add metadata outside of their CRDs to indicate what needs to be extracted for binding.
+
+## Improvements
+
+Validate advanced volume mounting support for projecting sensitive information as "files".
+
+## Duck type in backing service resources
+
+
+Identify the information that a duck typed resource needs to contain and 
+support binding secret generation based on the same.
```

Comment:
Sure, if a resource contains `.status.binding` field which refers to a secret, it would get picked up. This is something that the spec is bringing in :)